### PR TITLE
header component in one merged commit

### PR DESCRIPTION
**Pull request by ochanje210**
This is header component with squashed commits as one

### CODE REVIEW

**Comment by fguy** [comment]
Please revert src/test directory back. https://facebook.github.io/jest/docs/api.html#testpathdirs-array-string

File: gulpfile.babel.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ gulp.task('build:server', ['clean:server'], () => compileNodeJS('src/{server,sha
 
 gulp.task('build', ['build:client', 'build:server', 'build:static']);
 
-gulp.task('build:test-client', ['clean:test'], () => compileClientJS(['./src/test/browser/index.js'], 'index.js', './dist-test/test/browser'));
+//gulp.task('build:test-client', ['clean:test'], () => compileClientJS(['./src/__tests__/browser/index.js'], 'index.js', './dist-test/test/browser'));
```
Comment (fguy) [inline gulpfile.babel.js]: Why did you get rid of build:test-client?

Comment (ochanje210) [reply gulpfile.babel.js]: oops it was a mistake!

File: src/shared/components/routes.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ import MapLayout from '../components/MapLayout';
 export default (
   <Route path="/" component={AppContainer}>
     <IndexRoute component={IndexContainer}/>
-    <Route path="map" component={MapLayout}/>
+    <Route path="map" component={AppContainer}>
```
Comment (ovekyc) [inline src/shared/components/routes.js:11]: AppContainter is duplicated

Comment (ochanje210) [reply src/shared/components/routes.js:11]: well it is because currently our map component is being routed to /map.
if we use "/" path for the map component, then we don't have to use it twice.

File: gulpfile.babel.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ gulp.task('build:test-json', ['clean:test'], () => gulp.src('src/**/*.json')
   .pipe(gulp.dest('./dist-test/'))
 );
 
-gulp.task('build:test', ['build:test-client', 'build:test-server', 'build:test-json']);
+gulp.task('build:test', ['build:test-server', 'build:test-json']);
```
Comment (mun625) [inline gulpfile.babel.js]: why did you delete test-client?
I saw it previous comment, but still left. Is there any purpose?

Comment (ochanje210) [reply gulpfile.babel.js]: @mun625 It is because I deleted unit test and on line number 83, I added 'jest' which does client-side test already. Also, test/browser/index.js no longer exists..... good point I do not need line 37 actually.

**Comment by ochanje210** [comment]
@fguy I reverted!

File: gulpfile.babel.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ gulp.task('build:server', ['clean:server'], () => compileNodeJS('src/{server,sha
 
 gulp.task('build', ['build:client', 'build:server', 'build:static']);
 
-gulp.task('build:test-client', ['clean:test'], () => compileClientJS(['./src/test/browser/index.js'], 'index.js', './dist-test/test/browser'));
+gulp.task('build:test-client', ['clean:test'], () => compileClientJS(['./src/__tests__/browser/index.js'], 'index.js', './dist-test/test/browser'));
```
Comment (fguy) [inline gulpfile.babel.js]: **tests** directory is not acceptable.

File: gulpfile.babel.js (hunk 3 above)
Comment (fguy) [reply gulpfile.babel.js]: jest is not a good build task name. Please define it as **run:test-client**.

File: gulpfile.babel.js [hunk 5]
```diff
@@ -72,9 +72,16 @@ gulp.task('run:jsonlint', () => gulp.src(['**/*.json', '!node_modules/**'])
   .pipe(jsonlint.failAfterError())
 );
 
-gulp.task('run:test', ['build:test'], () => {
+gulp.task('jest', function() {
+  return gulp.src('src/**/__tests__/browser').pipe(jest({
```
Comment (fguy) [inline gulpfile.babel.js]: Below would run the task without **tests** directory.

gulp.task('run:test-client', ['build:test'], function() {
  return gulp.src('dist-test').pipe(jest({
    config: {
      rootDir: 'dist-test/test/browser',
      testRegex: '/index\.js'
    },
    unmockedModulePathPatterns: ['react']
  }));
});

File: src/test/browser/index.js [hunk 6]
```diff
@@ -1,6 +0,0 @@
-// Entry point for tests that will be run in Electron (browser).
```
Comment (fguy) [inline src/test/browser/index.js:1]: Needs an index file as a test suite.

File: src/shared/components/routes.js (hunk 2 above)
Comment (ovekyc) [reply src/shared/components/routes.js:11]: OH i got it

**Comment by ochanje210** [comment]
now it is using enzyme with code and lab instead of Jest

**Comment by fguy** [comment]
Please do not copy and paste. Wait until his job done and rebase.

**Comment by ochanje210** [comment]
oops ok. I only copied, .setup.js and settings in gulpfile

File: package.json [hunk 7]
```diff
@@ -26,6 +26,11 @@
   ],
   "dependencies": {
     "axios": "^0.14.0",
+    "babel-plugin-transform-class-properties": "^6.11.5",
```
Comment (fguy) [inline package.json:29]: I don't think these 'babel-*' are valid dependencies.

File: package.json [hunk 8]
```diff
@@ -36,6 +41,7 @@
     "fb": "^1.1.1",
     "google-map-react": "^0.18.3",
     "gulp": "^3.9.1",
+    "gulp-jest": "^0.5.2",
```
Comment (fguy) [inline package.json:44]: Remove

Comment (ochanje210) [reply package.json:44]: urg, I will remove
